lstchain/scripts/lstchain_dvr_pixselector.py

```diff
@@ -0,0 +1,516 @@
+decide the threshold subrun by subrun.
+
+The approach is the following: we run first the script over *all* subruns of a
+run, for example:
+
+lstchain_dvr_pixselector -f "/.../dl1_LST-1.Run12469.????.h5"
+
+(the default "action" compute_dvr_settings will be chosen by default, and the
+default parameters will be used for "picture_threshold" and "number_of_rings" -
+see below)
+This creates in the output directory (which is the current by default) a file
+DVR_settings_LST-1.Run12469.h5 which contains a table "run_summary" which
+includes the DVR algorithm parameters determined for each subrun.
+
+Then we run again the script over all subruns, and using the option to create
+the pixel maks:
+
+lstchain_dvr_pixselector -f "/.../dl1_LST-1.Run12469.????.h5" --action create_pixel_masks
+
+The script will detect that the file DVR_settings_LST-1.Run12469.h5 already
+exists, read it, and use, as threshold for DVR, the average for all subruns,
+in p.e., rounded to the closest lower integer (we do not want to have too many
+different ways of reducing the data, so we "discretize" the threshold in
+1-p.e. steps). Then the event-wise pixel maks (selected pixels) for the subrun
+will be computed and written out to a file
+Pixel_selection_LST-1.Run12469.xxxx.h5  for each subrun xxxx
+
+Note that when the option --action create_pixel_masks is used, the options
+--number-of-rings and --picture-threshold are ignored, since in that
+case the DVR settings will be obtained from the previously created DVR_settings
+file.
+
+"""
+
+import argparse
+from pathlib import Path
+import tables
+import glob
+import os
+import numpy as np
+import time
+
+from lstchain.paths import parse_dl1_filename
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.io import read_table
+from ctapipe.core import Container, Field
+from ctapipe.io import HDF5TableWriter
+from ctapipe.containers import EventType
+
+parser = argparse.ArgumentParser(description="DVR pixel selector")
+
+parser.add_argument('-f', '--dl1-files', dest='dl1_files',
+                    type=str, default='',
+                    help='Input DL1 file names')
+parser.add_argument('-o', '--output-dir', dest='output_dir',
+                    type=Path, default='./',
+                    help='Path to the output directory (default: %(default)s)')
+parser.add_argument('-t', '--picture-threshold', dest='picture_threshold',
+                    type=float, default=8.,
+                    help='Picture threshold (p.e., default: %(default)s)')
+parser.add_argument('--action', dest='action', type=str,
+                    default='compute_dvr_settings',
+                    help='compute_dvr_settings (default) or create_pixel_masks')
+parser.add_argument('-n', '--number-of-rings', dest='number_of_rings',
+                    type=int, default=1,
+                    help='Number of rings around picture pixels (default: %('
+                         'default)s)')
+
+class PixelMask(Container):
```

Comment:
shouldn't we define all these containers elsewhere? I mean, not in the script that is running the DVR

Reply:
It is not to be used anywhere else in lstchain. The program which reads the files in is necessarily a C++ program, outside of lstchain.